avro/src/main/java/com/fasterxml/jackson/dataformat/avro/AvroAnnotationIntrospector.java

```diff
@@ -161,6 +162,10 @@ public Object findSerializer(Annotated a) {
     }
 
     protected TypeResolverBuilder<?> _findTypeResolver(MapperConfig<?> config, Annotated ann, JavaType baseType) {
+        // If there's a custom deserializer, then it's responsible for handling any type information
+        if (config.getAnnotationIntrospector().findDeserializer(ann) != null) {
```

Comment:
I don' t think this approach makes sense: annotations are just one way to register deserializers; modules register others. But beyond this, as I said, type handlers and content handlers are not meant to be aware of each other.